proposals/user-scripts-execute-api.md

```diff
@@ -0,0 +1,150 @@
+# Extension API Proposal: userScripts.execute()
+
+## Background
+
+### User Scripts and User Scripts Managers
+
+User scripts are (usually relatively small) snippets of code that are injected into web pages in order to modify the page's appearance or behavior. User script managers are a type of extension that is used to manage the collection and injection of these user scripts, determining when and how to inject them on web pages.
+
+### Manifest V3
+
+Manifest Version 3 (MV3) restricts the ability for extensions to use remotely-hosted code since it is a major security risk (stores cannot review or audit code that isn't in the extension package). This is directly at odds with user script managers, which fundamentally require the execution of arbitrary code (the user scripts themselves). This created a feature gap between MV2 and MV3.
+
+### User Scripts API
+
+The User Scripts API was introduced in MV3 to register user scripts with arbitrary code, specifically targeting user script managers' use cases. User scripts are injected based on url matching through match patterns and globs. The API requires a new permission, and allows each browser to implement their own restrictions.
+In the User Script API proposal, executing a user script one time was [presented](https://github.com/w3c/webextensions/blob/main/proposals/user-scripts-api.md#execute-user-scripts-one-time) as a potential future enhancement.
+
+## API Overview
+
+### Objective
+
+Allow extensions to inject user scripts programmatically into web content given a target context.
+
+#### Use Cases
+
+The user wants to inject a user script on the page currently loaded in a given tab one time, but does not want to inject the script on that site automatically in the future. That is, the parameter that determines the injection is a tab ID and not based on origin.
+
+### Availability: Public
+
+This API will be public to all extensions (with the appropriate permission).
+
+### Consumers
+
+Any existing user script manager consumers of `chrome.tabs.executeScript()` that inject user scripts programmatically are intended to migrate to this new version in Manifest V3.
+
+## Implementation
+
+### API Schema
+
+#### Types
+
+```
+dictionary UserScriptInjection {
+  // Whether the injection should be triggered in the target as soon as
+  // possible. Note that this is not a guarantee that injection will occur
+  // prior to page load, as the page may have already loaded by the time the
+  // script reaches the target.
+  injectImmediately?: boolean,
+  // The script source to inject into the target.
+  js: ScriptSource,
+  // Details specifying the target into which to inject the script.
+  target: InjectionTarget,
+  // The JavaScript "world" to run the script in. The default is `USER_SCRIPT`.
+  world?: ExecutionWorld
```

Comment:
Missing comma at end. This is also the case at many of the other types. Let's add them.